hack/update-istio.sh

```diff
@@ -19,37 +19,89 @@ set -euo pipefail
 SLEEP_TIME=10
 VERSIONS_YAML_FILE=${VERSIONS_YAML_FILE:-"versions.yaml"}
 
-COMMIT=$(yq '.versions[] | select(.name == "latest") | "git ls-remote --heads " + .repo + ".git " + .branch + " | cut -f 1"' "${VERSIONS_YAML_FILE}" | sh)
-CURRENT=$(yq '.versions[] | select(.name == "latest") | .commit' "${VERSIONS_YAML_FILE}")
-
-if [ "${COMMIT}" == "${CURRENT}" ]; then
-  echo "${VERSIONS_YAML_FILE} is already up-to-date with latest commit ${COMMIT}."
-  exit 0
-fi
-
-echo Updating version 'latest' to commit "${COMMIT}"
-echo "Verifying the artifacts are available on GCS, this might take a while - you can abort the wait with CTRL+C"
-
-URL="https://storage.googleapis.com/istio-build/dev/${COMMIT}"
-until curl --output /dev/null --silent --head --fail "${URL}"; do
-    echo -n '.'
-    sleep ${SLEEP_TIME}
-done
-echo
-
-FULL_VERSION=$(curl -sSfL "${URL}")
-echo Full version: "${FULL_VERSION}"
-
-PARTIAL_VERSION="${FULL_VERSION%.*}"
-echo Partial version: "${PARTIAL_VERSION}"
-
-yq -i '
-    (.versions[] | select(.name == "latest") | .version) = "'"${PARTIAL_VERSION}"'" |
-    (.versions[] | select(.name == "latest") | .commit) = "'"${COMMIT}"'" |
-    (.versions[] | select(.name == "latest") | .charts) = [
-        "https://storage.googleapis.com/istio-build/dev/'"${FULL_VERSION}"'/helm/base-'"${FULL_VERSION}"'.tgz",
-        "https://storage.googleapis.com/istio-build/dev/'"${FULL_VERSION}"'/helm/cni-'"${FULL_VERSION}"'.tgz",
-        "https://storage.googleapis.com/istio-build/dev/'"${FULL_VERSION}"'/helm/gateway-'"${FULL_VERSION}"'.tgz",
-        "https://storage.googleapis.com/istio-build/dev/'"${FULL_VERSION}"'/helm/istiod-'"${FULL_VERSION}"'.tgz",
-        "https://storage.googleapis.com/istio-build/dev/'"${FULL_VERSION}"'/helm/ztunnel-'"${FULL_VERSION}"'.tgz"
-    ]' "${VERSIONS_YAML_FILE}"
+# Add a new entry in versions.yaml file. Only argument is the version (e.g. 1.22.5)
+function add_stable_version() {
+    template=$(cat <<-END
+{
+  "name": "v${1}",
+  "version": "${1}",
+  "repo": "https://github.com/istio/istio",
+  "commit": "${1}",
+  "charts": [
+    "https://istio-release.storage.googleapis.com/charts/base-${1}.tgz",
+    "https://istio-release.storage.googleapis.com/charts/istiod-${1}.tgz",
+    "https://istio-release.storage.googleapis.com/charts/gateway-${1}.tgz",
+    "https://istio-release.storage.googleapis.com/charts/cni-${1}.tgz",
+    "https://istio-release.storage.googleapis.com/charts/ztunnel-${1}.tgz"
+    ]
+}
+END
+    )
+    yq -i ".versions += ${template}" "${VERSIONS_YAML_FILE}"
```

Comment:
This adds the version to the end of the list. Since the version order is important, this should add the version in the correct spot.

Reply:
hmmm, ok. Why does the order matter? What would be the correct order?

Reply:
https://github.com/istio-ecosystem/sail-operator/blob/main/versions.yaml#L11

+ the order in versions.yaml is the order in which the versions appear in drop-downs and other lists.

Reply:
we need the version.yaml in order because we used in the e2e test and if we add a upgrade e2e test will be very useful also to have it in order. If not we will need to do some extra steps to order the versions before to start making the updates

Reply:
maybe we can add a comment to versions.yaml that explains that/why correct order is necessary? or we explain it in a markdown file somewhere

Reply:
done